pkg/controller/machine/controller.go

```diff
@@ -426,9 +426,18 @@ func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, er
 		machine.Status.LastUpdated = &now
 		if phase == phaseFailed && errorMessage != "" {
 			machine.Status.ErrorMessage = &errorMessage
+			if machine.Annotations == nil {
+				machine.Annotations = map[string]string{}
+			}
+			machine.Annotations[MachineInstanceStateAnnotationName] = "Unknown"
```

Comment:
I think we should only set unknown here https://github.com/openshift/machine-api-operator/pull/575/files#diff-b96ce5bf8686fb02d855188000ae1cb4R299-R304

Reply:
Here is another place, where the machine will get into Failed pahse, and the VM would not be monitored/updated anymore: https://github.com/openshift/machine-api-operator/blob/46410db09aae40009c72d3070ff7dc0bce6c4478/pkg/controller/machine/controller.go#L313-L316 I was thinking to handle it generically.

Reply:
make sense, in the other scenario the provider actually have a chance to set a valid state as it goes failed so might good to leave it. Happy to hear others thoughts.